test/test_engine.js

```diff
@@ -1142,6 +1142,50 @@ function buildEngineTests() {
       const unitsAfterReplace2 = engine.getLastSpecifiedUnits("manufacture");
       assert.equal(unitsAfterReplace2, "units");
     });
+
+    QUnit.test("initial charge considers last specified units", function (assert) {
+      const engine = new Engine(1, 3);
+
+      engine.setStanza("default");
+      engine.setApplication("TestApp");
+      engine.setSubstance("TestSub");
+
+      // Set equivalencies
+      engine.equals(new EngineNumber(1, "tCO2e / kg"), new YearMatcher(null, null));
+      engine.equals(new EngineNumber(1, "kwh / kg"), new YearMatcher(null, null));
+
+      // Set prior equipment
+      engine.setStream("priorEquipment", new EngineNumber(1000, "units"));
+
+      // Test 1: Set manufacture in kg, then set initial charge
+      engine.setStream("manufacture", new EngineNumber(100, "kg"));
+      engine.setInitialCharge(new EngineNumber(1, "kg / unit"), "manufacture");
+
+      const equipment1 = engine.getStream("equipment");
+      // With 100 kg manufacture, 1 kg/unit initial charge:
+      // New units = 100 / 1 = 100 units
+      // Total equipment = 1000 + 100 = 1100 units
+      assert.equal(equipment1.getValue(), 1100);
```

Comment:
If a call to recharge is added prior to this as requested, will this be a little less than 1100 units since some kg goes to recharge.